src/Files.App/Helpers/AppLifecycle.cs

```diff
@@ -0,0 +1,360 @@
+					{
+						int Index = nullIndex;
+						byte[] truncatedBuffer = new byte[Index];
+						Array.Copy(buffer, 0, truncatedBuffer, 0, Index);
+						string bufferStr = Encoding.UTF8.GetString(truncatedBuffer);
+						List<string> TabsWithIDArgStrList = JsonSerializer.Deserialize<List<string>>(bufferStr);
+						TabsWithIDArgList = TabsWithIDArgStrList.Select(x => TabItemWithIDArguments.Deserialize(x)).ToList();
+					}
+					else
+					{
+						TabsWithIDArgList = new List<TabItemWithIDArguments>();
+					}
+				}
+			}
+			finally
+			{
+				await Task.CompletedTask;
+			}
+		}
+
+		//	Write TabsWithIDArgList to SharedMemory
+		protected static async Task WriteSharedMemory()
+		{
+			try
+			{
+				List<string> TabsWithIDArgStrList = TabsWithIDArgList.Select(x => x.Serialize()).ToList();
+				string bufferStr = JsonSerializer.Serialize(TabsWithIDArgStrList);
+				byte[] buffer = Encoding.UTF8.GetBytes(bufferStr);
+
+				SharedMemory = CheckSharedMemory(buffer.Length);
+
+				using (MemoryMappedViewAccessor accessor = SharedMemory.CreateViewAccessor())
+				{
+					byte[] bufferClear = new byte[accessor.Capacity];
+					accessor.WriteArray(0, bufferClear, 0, bufferClear.Length);
+					accessor.WriteArray(0, buffer, 0, buffer.Length);
+				}
+			}
+			finally
+			{
+				await Task.CompletedTask;
+			}
+		}
+
+		//	Add Tabs of this instance to TabsWithIDList
+		private static List<TabItemWithIDArguments> AddTabsWithID()
+		{
+			List<TabItemWithIDArguments> OtherTabsWithIDArgList = TabsWithIDArgList.FindAll(x => x.InstanceID != InstanceID).ToList();
+			List<string> ThisInstanceTabsStr = MainPageViewModel.AppInstances.DefaultIfEmpty().Select(x => x.TabItemArguments.Serialize()).ToList();
+
+			List<TabItemWithIDArguments> ThisInstanceTabsWithIDArgList = ThisInstanceTabsStr.Select(x => TabItemWithIDArguments.Deserialize(x)).ToList();
+			List<TabItemWithIDArguments> NewTabsWithIDArgList = OtherTabsWithIDArgList.ToList();
+			NewTabsWithIDArgList.AddRange(ThisInstanceTabsWithIDArgList);
+			return NewTabsWithIDArgList;
+		}
+		private static List<TabItemWithIDArguments> RemoveTabsWithID()
+		{
+			List<TabItemWithIDArguments> OtherTabsWithIDArgList = TabsWithIDArgList.FindAll(x => x.InstanceID != InstanceID).ToList();
+			return OtherTabsWithIDArgList;
+		}
+		public static async Task UpDate()
+		{
+			await ReadSharedMemory();
+			TabsWithIDArgList = AddTabsWithID();
+			await WriteSharedMemory();
+			await ReadSharedMemory();
+			userSettingsService.GeneralSettingsService.LastAppsTabsWithIDList = TabsWithIDArgList.Select(x => x.Serialize()).ToList();
+		}
+		public static async void RemoveThisInstanceTabs()
+		{
+			await ReadSharedMemory();
+			TabsWithIDArgList = RemoveTabsWithID().ToList();
+			await WriteSharedMemory();
+			userSettingsService.GeneralSettingsService.LastAppsTabsWithIDList = TabsWithIDArgList.Select(x => x.Serialize()).ToList();
+		}
+
+		//	Restore LastAppsTabs
+		public static bool RestoreLastAppsTabs()
```

Comment:
This method shares a lot of code with `public static bool RestoreLastAppsTabs(MainPageViewModel mainPageViewModel)` just below. The common code should be refactored into one function for readability.

Reply:
Do not hesitate to create some line spaces between some instructions to create some room, it helps with the reading.

Reply:
RestoreLastAppsTabs() is not used, so delete it.